pkg/evaluators/identity/api_key.go

```diff
@@ -26,18 +26,20 @@ type APIKey struct {
 	Name           string              `yaml:"name"`
 	LabelSelectors k8s_labels.Selector `yaml:"labelSelectors"`
 	Namespace      string              `yaml:"namespace"`
+	KeySelectors   []string            `yaml:"keySelectors"`
 
 	secrets   map[string]k8s.Secret
 	mutex     sync.RWMutex
 	k8sClient k8s_client.Reader
 }
 
-func NewApiKeyIdentity(name string, labelSelectors k8s_labels.Selector, namespace string, authCred auth.AuthCredentials, k8sClient k8s_client.Reader, ctx context.Context) *APIKey {
+func NewApiKeyIdentity(name string, labelSelectors k8s_labels.Selector, namespace string, keySelectors []string, authCred auth.AuthCredentials, k8sClient k8s_client.Reader, ctx context.Context) *APIKey {
 	apiKey := &APIKey{
 		AuthCredentials: authCred,
 		Name:            name,
 		LabelSelectors:  labelSelectors,
 		Namespace:       namespace,
+		KeySelectors:    append(keySelectors, apiKeySelector),
```

Comment:
Sounds good 👍 

On a related note, what's your thoughts on instead of doing this `append`, use the kubebuilder default annotation in the API struct instead? The `append` works but this default value is hidden from the CRD, but not sure is this want we want in general
```
	// +kubebuilder:default:={"api_key"}
```